tests/30rooms/06invite.pl

```diff
@@ -154,3 +157,76 @@ sub invited_user_can_reject_invite
       Future->done(1);
    });
 }
+
+test "Invited user can see room metadata",
+   requires => [ local_user_and_room_fixtures(), local_user_fixture() ],
+
+   do => sub {
+      my ( $creator, $room_id, $invitee ) = @_;
+
+      my $state_in_invite;
+
+      Future->needs_all(
+         matrix_put_room_state( $creator, $room_id,
+            type => "m.room.name",
+            content => { name => "The room name" },
+         ),
+         matrix_put_room_state( $creator, $room_id,
+            type => "m.room.avatar",
+            content => { url => "http://something" },
+         ),
+      )->then( sub {
+         matrix_invite_user_to_room( $creator, $invitee, $room_id );
+      })->then( sub {
+         await_event_for( $invitee, sub {
+            my ( $event ) = @_;
+            return $event->{type} eq "m.room.member" &&
+                   $event->{room_id} eq $room_id;
+         });
+      })->then( sub {
+         my ( $event ) = @_;
+
+         require_json_keys( $event, qw( invite_room_state ));
+         my %state_by_type = map {
```

Comment:
I think this whole block would be clearer as a for loop rather than this sequence of fetches and maps which actually follows below.

Something along the lines of:

```perl:
my @futures = ();

foreach my $event_type (...) {
   push @futures, matrix_get_room_state( $creator, $room_id, type => $event_type )->then( sub {
      my ( $room_content ) = @_;
      my $invite_content = $event->{invite_room_state}->{$event_type};
      all { $room_content->{$_} eq $invite_content->{$_} } keys %$room_content, keys %$in_invite or
         die "Content does not match for event type $event_type"; # Or better yet, assert_hash_equals( $room_content, $invite_content )
   });
   Future->needs_all( $futures );
```